src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -978,16 +977,304 @@ void CmdSketcherViewSection::activated(int iMsg)
 
 bool CmdSketcherViewSection::isActive()
 {
-    Gui::Document *doc = getActiveGuiDocument();
-    if (doc) {
-        // checks if a Sketch Viewprovider is in Edit and is in no special mode
-        SketcherGui::ViewProviderSketch* vp = dynamic_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
-        if (vp /*&& vp->getSketchMode() == ViewProviderSketch::STATUS_NONE*/)
-            return true;
+    return isEditModeToolActive(getActiveGuiDocument());
+}
+
+/* Grid tool ------------------------------ Grid Activate */
+DEF_STD_CMD_A(CmdSketcherGridToggle)
+
+bool toggleGrid(Gui::Document* doc) {

```

Comment:
There is actually no need to make this a global function because it's only used within CmdSketcherCompGrid::activated.
And to get the current state you better query the corresponding QAction instead of getting it from the user parameter.